### PR TITLE
add `errors` docs page

### PR DESCRIPTION
**Pull request by phryneas**
This adds a new page for https://www.apollographql.com/docs/react/errors

For testing this locally, check out `https://github.com/apollographql/docs` in parallel to the `apollo-client` repo and run `npm run start:local -- ../apollo-client` - the page will be available at [local url](http://localhost:3000/errors#{%22version%22:%20%220.0.0-pr-10887-20230522141246%22,%20%22message%22:%208,%20%22args%22:%20[%22foo%22,%20%22bar%22]})

Or, in the [deployed preview link](https://deploy-preview-10909--apollo-client-docs.netlify.app/errors/#{%22version%22:%20%220.0.0-pr-10887-20230522141246%22,%20%22message%22:%208,%20%22args%22:%20[%22foo%22,%20%22bar%22]})

image

### CODE REVIEW

File: docs/shared/DisplayClientError.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    if (scriptRef.current.children.length === 0) {
+      const script = document.createElement('script');
+      script.type = 'module';
+      script.textContent =`
+        globalThis[Symbol.for("importInvariantErrorCodes")] = (version) => {
+          if (!version || !/^[0-9a-zA-Z.-]+$/.test(version)) {
+            throw 'Invalid Version!';
+          }
+          return import("https://cdn.jsdelivr.net/npm/@apollo/client@" + version + "/invariantErrorCodes.js");
+        }
```
Comment (phryneas) [inline docs/shared/DisplayClientError.js:58]: I want to point out this code.

It does load a user-specified Apollo Client version from an external CDN and execute the code in the file. 
We do check the `version` string against a Regex only allowing for characters, numbers and the `.` and `-` symbols, so any kind of XSS should be impossible unless an attacker could publish a compromised version of Apollo Client to that CDN - and then they had other, a lot more direct, ways of attacking people.

**Comment by phryneas** [comment]
Generally, this technically works - I would be grateful for any kind of wording corrections or styling suggestions here.

File: docs/source/errors.mdx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+You can call the `loadErrorMessages` function from `"@apollo/client/dev"` to
+load error messages.<br/>
+Similarly, you can use `loadDevMessages` to import development-time console
+messages.
+
+### Only shipping error messages in development builds
+
+In most environments, you can check for `__DEV__`
+to make sure that you are in a developement environment.
```
Comment (bignimbus) [inline docs/source/errors.mdx:24]:
```suggestion
to make sure that you are in a development environment.
```

File: docs/shared/DisplayClientError.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  }, []);
+  return (
+    <div>
+      <div ref={scriptRef}></div>
+      {error ? (
+        <>
+          <MDX.h3>
+            Error loading error codes
+          </MDX.h3>
+          <MDX.blockquote>{error}</MDX.blockquote>
```
Comment (bignimbus) [inline docs/shared/DisplayClientError.js:93]: This looks nice so I'm fine keeping it. I wonder whether we should try using a code block style instead?

**Review by bignimbus — COMMENTED** [review]
No concerns here. I left a couple small notes. Thanks so much for this awesome PR!

File: docs/shared/DisplayClientError.js (hunk 3 above)
Comment (phryneas) [reply docs/shared/DisplayClientError.js:93]: I think that would add a lot more visual fluff, like the "JavaScript" tab on the top.

**Comment by StephenBarlow** [comment]
Pushed a commit that's mostly reformatting so I can next open a PR that's entirely content changes

**Comment by phryneas** [comment]
Argh... the in-page anchor tags for the sections mess everything up there 🤦

**Comment by phryneas** [comment]
Okay, anchor tag links are now removed for this page only.

Also, I had to rework the script loading logic because it had a different timing in Safari. Should work everything now, and it's also a bit more React-y.

**Review by jerelmiller — APPROVED** [review]
🎉 🎉 Thanks for getting this done!

**Pull request merged**